.github/workflows/integration-tests.yml

```diff
@@ -143,26 +129,42 @@ jobs:
             ~/.triton/nvidia
             ~/.triton/pybind11
           key: ${{ runner.os }}-${{ runner.arch }}-llvm-${{ steps.cache-key.outputs.llvm }}-nvidia-${{ steps.cache-key.outputs.nvidia }}-pybind11-${{ steps.cache-key.outputs.pybind11 }}
-
+      - # Cache ~/.triton/cache because the vast majority of unit test time is
+        # spent compiling.  Triton won't (well, should not) use these cached files
+        # if something internal to Triton changes, because Triton's internal
+        # source code is part of the cache key.
+        #
+        # Similarly, cache ~/.cache/ccache to speed up compilation.
+        #
+        # On branch `main` we always start from an empty cache, i.e. we skip the
+        # "restore" step.  This is to prevent the caches from accumulating stale
+        # files over time.
+        name: Restore cache of ccache and Triton compilation artifacts
+        if: github.ref != 'refs/heads/main'
```

Comment:
Maybe check push event `github.event_name != 'push'` here? That way it's a bit flexible--if later we decided to enable on some other branches it will be just changing Line 12. Not sure whether we have such a need though; so minor issue.